### ISSUE TITLE
Appium install returns a non-zero exit code with NPM7

### ISSUE THREAD

**Issue by imbeyondboredom** [appium/appium#14901]
## The problem

Appium fails to install or at least returns a non-zero exit code while installing

## Environment

* Appium version (or git revision) that exhibits the issue: 1.19.0
* Last Appium version that did not exhibit the issue (if applicable): Unsure - maybe 1.11.0
* Desktop OS/version used to run Appium: OSX
* Node.js version (unless using Appium.app|exe): v15.2.1
* Npm or Yarn package manager: npm@7.0.10
* Mobile platform/version under test: N/A
* Real device or emulator/simulator: N/A
* Appium CLI or Appium.app|exe: CLI

## Details

When running the following commands on a fresh OS Catalina install:
```
brew install node
npm install -g appium
```
it errors with
```
npm ERR! code ENOENT
npm ERR! syscall chmod
npm ERR! path /usr/local/lib/node_modules/appium/node_modules/.bin/authorize-ios
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, chmod '/usr/local/lib/node_modules/appium/node_modules/.bin/authorize-ios'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 
```

However - if I go to run appium it does appear to launch. We're installing appium through Packer so the fact that it errors but is still installed is not ideal. I'm also not 100% convinced the install is complete because of this error but I can't be sure.

Someone else saw this same behavior [here](https://stackoverflow.com/questions/64655932/bin-authorize-ios-not-found-when-installing-appium-using-npm-install-g-appium)

## Other notable things:
- Since we're doing this with packer I can say that the OS is new and the test is repeatable
- I tried downgrading node but that does not seem to have an effect until v1.11.0. It also doesn't seem like I can update it to the latest

**Comment by KazuCocoa** [comment]
I was able to install the `authorize-ios` with appium 1.19.0 on my local via `npm install -g`. I guess npm failed to create the file in the installation process because of write permission.

**Comment by mykola-mokhnach** [comment]
Closed as not an issue

**Comment by imbeyondboredom** [comment]
I agree with closing this issue as it does appear to be NPM but wanted to post back here in case someone else stumbles across this problem.

This is fixed by installing the latest version of npm or 6.14.9
```
npm i -g npm@latest
npm install -g appium
```

**Comment by chrmoritz** [comment]
This issue is indeed about appium being unable to be installed with any `npm@7` version and the only workaround for it is to downgrade to `npm@6` (with `npm i -g npm@latest`).

It is caused by the unsupported way appium tries to create a `bin` symlink to `authorize-ios` in:

https://github.com/appium/appium/blob/09437412c8e6ddfc73034cf304fa6a41c8ce5a54/package.json#L31

In `npm@6` all dependencies were installed first before trying to create `bin` symlinks, but `npm@7` changed the installation order and now tries to create the `bin` symlinks **before** installing any dependencies. This still works fine if you only use the `bin` field [in the supported way to refer to local files only](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#bin). But since appium here refers to a file in the `./node_modules` dependencies, which just doesn't exist yet, the creating of the symlink will fail and error out the installation.

A possible fix would be to just point the `authorize-ios` bin field to a local script, which just reexecutes `authorize-ios` inside  `./node_modules/.bin`.

**Comment by davidshq** [comment]
FYI: `npm i -g npm@latest` now installs npm 7. To get six an individual will need to do `npm i -g npm@6`.

**Comment by chrmoritz** [comment]
This should be fixed for npm 7 starting with appium v1.20.0+ too.
